conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureHandler.java

```diff
@@ -32,48 +32,27 @@
-import java.util.function.BiFunction;
 import java.util.stream.Collectors;
 
 /**
  * Conjure routing mechanism which can be registered as an Undertow {@link HttpHandler}.
- * This handler takes care of exception handling, tracing, and web-security headers
- * among {@link ConjureHandler#WRAPPERS other features}.
+ * This handler takes care of exception handling, tracing, and web-security headers.
  */
 public final class ConjureHandler implements HttpHandler {
 
-    private static final ImmutableList<BiFunction<Endpoint, HttpHandler, HttpHandler>> WRAPPERS =
-            ImmutableList.<BiFunction<Endpoint, HttpHandler, HttpHandler>>of(
-            // Allow the server to configure UndertowOptions.DECODE_URL = false to allow slashes in parameters.
-            // Servers which do not configure DECODE_URL will still work properly except for encoded slash values.
-            // When DECODE_URL has not been disabled, the following handler will no-op
-            (endpoint, handler) -> new URLDecodingHandler(handler, "UTF-8"),
-            // no-cache and web-security handlers add listeners for the response to be committed,
-            // they can be executed on the IO thread.
-            (endpoint, handler) -> Methods.GET.equals(endpoint.method())
-                    // Only applies to GET methods
-                    ? new NoCachingResponseHandler(handler) : handler,
-            (endpoint, handler) -> new WebSecurityHandler(handler),
-            // It is vitally important to never run blocking operations on the initial IO thread otherwise
-            // the server will not process new requests. all handlers executed after BlockingHandler
-            // use the larger task pool which is allowed to block. Any operation which sets thread
-            // state (e.g. SLF4J MDC or Tracer) must execute on the blocking thread otherwise state
-            // will not propagate to the wrapped service.
-            (endpoint, handler) -> new BlockingHandler(handler),
-            // Logging context and trace handler must execute prior to the exception
-            // to provide user and trace information on exceptions.
-            (endpoint, handler) -> new LoggingContextHandler(handler),
-            (endpoint, handler) -> new TracedOperationHandler(
-                    handler, endpoint.method() + " " + endpoint.template()),
-            (endpoint, handler) -> new ConjureExceptionHandler(handler)
-    ).reverse();
-
+    private final EndpointHandlerWrapper stackedEndpointHandlerWrapper;
     private final RoutingHandler routingHandler;
 
-    private ConjureHandler(HttpHandler fallback, List<Endpoint> endpoints) {
+    private ConjureHandler(
+            HttpHandler fallback,
+            List<Endpoint> endpoints,
+            EndpointHandlerWrapper stackedEndpointHandlerWrapper) {
         this.routingHandler = Handlers.routing().setFallbackHandler(fallback);
+        this.stackedEndpointHandlerWrapper = stackedEndpointHandlerWrapper;
```

Comment:
Can we avoid providing the wrapper list to the ConjureHandler, and instead wrap endpoints in `build()`?

Reply:
done